src/runtime/cpu_features.h

```diff
@@ -0,0 +1,54 @@
+#ifndef HALIDE_CPU_FEATURES_H
+#define HALIDE_CPU_FEATURES_H
+
+#include "HalideRuntime.h"
+#include "cpu_features.h"
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+// Return two masks:
+// One with all the CPU-specific features that might possible be available on this architecture ('known'),
+// and one with the subset that are actually present ('available').
+struct CpuFeatures {
+    static const int kWordCount = (halide_target_feature_end + 63) / (sizeof(uint64_t) * 8);
+
+    __attribute__((always_inline))
+    void set_known(int i) {
+        known[i / 64] |= ((uint64_t) 1) << (i % 64);
```

Comment:
(i / 64) isn't going to optimize to `i>>6` for signed ints. Better to use explicit `>>6` and `&63` here and below.

Reply:
done